extension/src/popup/views/AccountHistory/index.tsx

```diff
@@ -0,0 +1,201 @@
+            ? ` + ${operationCount - 1} ops`
+            : null}
+          <div className="AccountHistory__date">
+            {isRecipient ? "Received" : "Sent"} • {date}
+          </div>
+        </div>
+
+        <div>{renderPaymentComponent()}</div>
+      </div>
+    </div>
+  );
+};
+
+enum SELECTOR_OPTIONS {
+  ALL = "ALL",
+  SENT = "SENT",
+  RECEIVED = "RECEIVED",
+}
+
+type HistorySegments = {
+  [key in SELECTOR_OPTIONS]: HorizonOperation[] | [];
+};
+
+export const AccountHistory = () => {
+  const publicKey = useSelector(publicKeySelector);
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const [selectedSegment, setSelectedSegment] = useState(SELECTOR_OPTIONS.ALL);
+  const [historySegments, setHistorySegments] = useState({
+    [SELECTOR_OPTIONS.ALL]: [],
+  } as HistorySegments);
+
+  const STELLAR_EXPERT_URL = `https://stellar.expert/explorer/${
+    networkDetails.isTestnet ? "testnet" : "public"
+  }`;
+
+  useEffect(() => {
+    const createRemainingSegments = (operations: HorizonOperation[]) => {
+      const segments = {
+        [SELECTOR_OPTIONS.SENT]: [] as HorizonOperation[],
+        [SELECTOR_OPTIONS.RECEIVED]: [] as HorizonOperation[],
+      };
+      operations.forEach((operation) => {
+        if (isPayment(operation.type)) {
+          if (operation.source_account === publicKey) {
+            segments[SELECTOR_OPTIONS.SENT].push(operation);
+          }
+          if (operation.to === publicKey) {
+            segments[SELECTOR_OPTIONS.RECEIVED].push(operation);
+          }
+        }
+      });
+
+      return segments;
+    };
+
+    const fetchAccountHistory = async () => {
+      try {
+        const res = await getAccountHistory({ publicKey, networkDetails });
+        const paymentSegments = createRemainingSegments(res.operations);
+        setHistorySegments({
+          [SELECTOR_OPTIONS.ALL]: res.operations,
+          ...paymentSegments,
+        });
+      } catch (e) {
+        console.error(e);
+      }
+    };
+    fetchAccountHistory();
+  }, [publicKey, networkDetails]);
+
+  return (
+    <div className="AccountHistory">
+      <PopupWrapper>
+        <header className="AccountHistory__header">Transactions</header>
+        <div className="AccountHistory__selector">
+          {Object.values(SELECTOR_OPTIONS).map((option) => (
+            <div
+              key={option}
+              className={`AccountHistory__selector__item ${
+                option === selectedSegment
+                  ? "AccountHistory__selector__item--active"
+                  : ""
+              }`}
+              onClick={() => setSelectedSegment(option)}
+            >
+              {option}
+            </div>
+          ))}
+        </div>
+        <ul className="AccountHistory__list">
+          {historySegments[SELECTOR_OPTIONS[selectedSegment]].map(
+            (operation: HorizonOperation) => (
+              <HistoryItem
+                key={operation.id}
```

Comment:
💭  thoughts on passing to `HistoryItem` the type of payment? So you don't have to check again in that component.

Also a (silly) edge case was my account had a couple payments to itself. So the icon was a "receive" icon even though it was in the send payment group (which I think makes sense it's in both send and receive)

Reply:
I addressed the first part of your comment.

Good point about self payments. There really isn't a good solve for this besides flipping the icon based on view, and that I think adds some unnecessary complexity for a pretty small edge case. I'm gonna pass on that one for now and see if it ends up being an issue in the future!